app/service/model/date.py

```diff
@@ -116,3 +116,15 @@ def check_present_year(db: Session, year: int):
         from app.models.date import Date
 
         return True if db.query(Date).filter_by(year=year).first() else False
+
+    @staticmethod
+    @session_hook
+    def get_current_day_from_record(db: Session):
+        from datetime import datetime
+        from app.models.date import Date
+
+        full_date = datetime.utcnow().date()
+
+        record = db.query(Date).filter_by(full_date=full_date).first()
+
+        return record if record else None
```

Comment:
This function should never return none it should always call the run method to create date record if current day doesn't  exist before executing it logic